src/units/combine.rs

```diff
@@ -215,29 +232,38 @@ mod test {
             }
         ).unwrap();
 
-        // Set all units to stalled, check that the target goes stalled.
-        join!(u1.status(Stalled), u2.status(Stalled), u3.status(Stalled));
-        assert_eq!(t.recv().await, Err(Stalled));
+        // Set one unit to stalled, this triggers picking a new source healthy
+        // with data but there isn’t one, so we go stalled.
+        u1.send_stalled().await;
+        t.recv_stalled().await.unwrap();
+
+        // Now stall the other ones. That shouldn’t change anything.
+        // the last one.
```

Comment:
What is this "// the last one." comment doing here? It seems to be orphaned/unrelated to the surrouding text.